src/main/java/core/StandardForwardModel.java

```diff
@@ -0,0 +1,24 @@
+package core;
+
+import core.actions.AbstractAction;
+import utilities.Utils;
+
+public abstract class StandardForwardModel extends AbstractForwardModel {
```

Comment:
I like this, a neat way to handle the multiple steps needed to be taken care of in next(). I'd propose expanding this even further, as there are 4 things done in most games in the next() function (that sometimes it's easy to forget about, so breaking it down like this and making it mandatory to implement is helpful):
- (done) Apply action
- (done) Execute any other required game rules post-action (e.g. change the phase of the game);
- Check for game end, and process the end of game appropriately
- Check if the player's turn or round needs to end, and do it

I'm not sure how much variety there is now in how these steps are done, but there are a lot of games that do these 4 steps pretty consistently so I think they can all be abstracted up like the first 2 you've done already.

Reply:
A follow-up thought: I think there's also a lack of consistency in the games implemented as to which of these things happen on the action execution, or the turn order itself, or in the next() in the FM. Would be good to look into this at some point and see if we can standardize things more...

Reply:
"A follow-up thought: I think there's also a lack of consistency in the games implemented as to which of these things happen on the action execution, or the turn order itself, or in the next() in the FM. Would be good to look into this at some point and see if we can standardize things more..."

You are not kidding! In a few cases I've just had to leave existing games untouched as converting them to a cleaner format woudl be too dangerous (especially in the absence of good unit test coverage).  Very happy to extend further - and this should also tie in with the discussion on turn order. 

My thinking general thinking is that for backwards compatibility we have a new model way of implementing a game, but if you want to you can make a spaghetti mess in the old style. (Unless we want to commit to the work to re-implement the games that are problematic.)